kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
```

Comment:
At a bare minimum, we need a method that returns *All* IK solutions for a given pose.  For 6DOF robots, this is a fixed set of configurations (as you point out).  What this PR tries to address is how you control/configure the what *All* means when there is redundancy.  This requirement to be able to configure the IK is driven form the conversation on the MoveIt list.  That being said, we already have a use case within Descartes where a rough search is performed, followed by a fine search if required, so there is a need.